tools/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -434,6 +434,145 @@ object SQLPlanParser extends Logging {
+       }
+     }
+
+     (parsedExpressions.distinct.toArray, equiJoinSupportedTypes(buildSide, joinType))
+   }
+
+  def parseNestedLoopJoinExpressions(exprStr: String): (Array[String], Boolean) = {
+    // BuildRight, LeftOuter, ((CEIL(cast(id1#1490 as double)) <= cast(id2#1496 as bigint))
+    // AND (cast(id1#1490 as bigint) < CEIL(cast(id2#1496 as double))))
+    val parsedExpressions = ArrayBuffer[String]()
+    // Get joinType and buildSide by splitting the input string.
+    val nestedLoopParameters = exprStr.split(",", 3)
+    val buildSide = nestedLoopParameters(0).trim
+    val joinType = nestedLoopParameters(1).trim
+
+    if (nestedLoopParameters.size > 2) { // condition present on join columns
+      val exprSepAND = if (exprStr.contains("AND")) {
+        exprStr.split(" AND ").map(_.trim)
+      } else {
+        Array(exprStr)
+      }
+      val exprSplit = if (exprStr.contains(" OR ")) {
+        exprSepAND.flatMap(_.split(" OR ").map(_.trim))
+      } else {
+        exprSepAND
+      }
+
+      // Remove parentheses from the beginning and end to get only the expressions
+      val paranRemoved = exprSplit.map(_.replaceAll("""^\(+""", "").replaceAll("""\)\)$""", ")"))
+      val conditionalExprPattern = """([(\w# )]+) ([+=<>|]+) ([(\w# )]+)""".r
+      paranRemoved.foreach { case expr =>
+        if (expr.contains(" ")) {
+          conditionalExprPattern.findFirstMatchIn(expr) match {
+            case Some(func) =>
+              logDebug(s" found expr: $func")
+              if (func.groupCount < 3) {
+                logError(s"found incomplete expression - $func")
+              }
+              // get expressions from condition
+              val expressions = Array(func.group(1), func.group(3))
+              expressions.foreach { expr =>
+                // Add function name to result
+                val functionName = getFunctionName(functionPattern, expr)
+                functionName match {
+                  case Some(func) => parsedExpressions += func
+                  case _ => // NO OP
+                }
+              }
+            case None => logDebug(s"Incorrect expression - $expr")
+          }
+        } else {
+          // likely some function call, add function name to result
+          val functionName = getFunctionName(functionPattern, expr)
+          functionName match {
+            case Some(func) => parsedExpressions += func
+            case _ => // NO OP
+          }
+        }
+      }
+    } else {
+      // NO OP
+    }
+    (parsedExpressions.distinct.toArray, nestedLoopJoinSupportedTypes(buildSide, joinType))
+  }
+
+  private def isJoinTypeSupported(joinType: String): Boolean = {
+    joinType match {
+      case JoinType.Cross => true
+      case JoinType.Inner => true
+      case JoinType.LeftSemi => true
+      case JoinType.LeftOuter => true
+      case JoinType.RightOuter => true
+      case JoinType.LeftAnti => true
+      case _ => false
```

Comment:
what about FullOuter and ExistenceJoin here?  I thought we supported some of those. Are we not able to tell that here? if not we should add comments

Reply:
FullOuter is not supported for BroadcastNestedLoopJoin, and for equiJoins there is a caveat that it doesn't support for struct keys. I updated the code to mention it as supported for equiJoins. For ExistenceJoins, added comment that it doesn't show up in event logs.